### DISCUSSION TITLE
Supply - upload native symbols

### DISCUSSION THREAD

**Discussion by zyvpeople** [fastlane/fastlane#17517] [Ideas] [answered]
Hi!
Is it possible to add functionality for **Supply** tool to upload native debug symbols and not only ProGuard mapping (https://support.google.com/googleplay/android-developer/answer/9848633#upload_file)?
In the source code I found that Supply uploads only ProGuard mappings (https://github.com/fastlane/fastlane/blob/master/supply/lib/supply/client.rb#L335).
But Google Play Dev Api also provides a possibility to upload native debug symbols (https://developers.google.com/android-publisher/api-ref/rest/v3/edits.deobfuscationfiles).
Thanks.

**Comment by aberaud** [comment]
Would be very useful indeed. Without this, debugging native crashes from the Play Store is a nightmare.

**Comment by rogerluan** [comment]
That sounds wonderful! 

And by taking a quick glance at the linked resources that you posted, it doesn't seem to be too complicated to implement this. We'd need to add new options to _supply_ (in `supply/lib/supply/options.rb`) and read them at the place you posted (https://github.com/fastlane/fastlane/blob/master/supply/lib/supply/client.rb#L335) 🤗 

Contributions are always welcome, so let me know if you'd like to work on this @zyvpeople ! Feel free to tag me to review your code if you open a PR 🙌

**Comment by MarioNoll** [comment]
Duplicate of #17270

**Comment by capyvara** [comment, marked as answer]
I'll try do do it.

**Reply by capyvara** [reply]
It's there #18526, thumbs up so they merge it quickly. =)

**Reply by rogerluan** [reply]
It was just merged! 🎉 thanks for your contribution!

**Reply by ShivamJoker** [reply]
@capyvara I don't see this on fastlane website docs? Can you please tell me how to add the path in fastfile? Instead of passing in in args
